src/components/GoodsList/GoodsList.tsx

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+
+import './GoodsList.scss';
+
+type Props = {
+  goodsFromServer: string[],
+};
+
+type State = {
+  goods: string[]
+  isVisibility: boolean,
+  isReverse: boolean,
+  sortBy: string,
+  goodsLength: number,
+};
+
+export class GoodsList extends React.Component<Props, State> {
+  state = {
+    goods: this.props.goodsFromServer,
+    isVisibility: false,
+    isReverse: false,
+    sortBy: '',
+    goodsLength: 1,
+  };
+
+  isVisibility = () => {
+    this.setState(state => (
+      {
+        isVisibility: !state.isVisibility,
+      }
+    ));
+  };
+
+  reverse = () => {
+    this.setState(state => (
+      {
+        isReverse: !state.isReverse,
+      }
+    ));
+  };
+
+  reset = () => {
+    this.setState({
+      goods: this.props.goodsFromServer,
+      isReverse: false,
+      sortBy: '',
+      goodsLength: 1,
+    });
+  };
+
+  sortByName = () => {
+    this.setState({
+      sortBy: 'name',
+    });
+  };
+
+  sortByLength = () => {
+    this.setState({
+      sortBy: 'length',
+    });
+  };
+
+  render() {
+    const {
+      goods,
+      isVisibility,
+      isReverse,
+      sortBy,
+      goodsLength,
+    } = this.state;
+
+    const goodsList = [...goods].filter(good => good.length >= goodsLength);
```

Comment:
```suggestion
    const goodsList = goods.filter(good => good.length >= goodsLength);
```
Filter не мутирующий метод, зачем делать копию?